core/new-gui/src/app/dashboard/component/login/login.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, OnInit } from "@angular/core";
+import { UserService } from "../../../common/service/user/user.service";
+import { FormBuilder, FormControl, FormGroup, Validators } from "@angular/forms";
+import { isDefined } from "../../../common/util/predicate";
+import { filter } from "rxjs/operators";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+
+import { Router } from "@angular/router";
+
+@UntilDestroy()
+@Component({
+  selector: "texera-login",
+  templateUrl: "./login.component.html",
+  styleUrls: ["./login.component.scss"],
+})
+export class LoginComponent implements OnInit {
+  public loginErrorMessage: string | undefined;
+  public registerErrorMessage: string | undefined;
+  public allForms: FormGroup;
+
+  constructor(private formBuilder: FormBuilder, private userService: UserService, public router: Router) {
+    this.allForms = this.formBuilder.group({
+      loginUsername: new FormControl("", [Validators.required]),
+      registerUsername: new FormControl("", [Validators.required]),
+      loginPassword: new FormControl("", [Validators.required, Validators.minLength(6)]),
+      registerPassword: new FormControl("", [Validators.required, Validators.minLength(6)]),
+      registerConfirmationPassword: new FormControl("", [Validators.required, this.confirmationValidator]),
+    });
+  }
+
+  ngOnInit() {
+    this.detectUserChange();
+  }
+
+  public updateConfirmValidator(): void {
+    // immediately update validator (asynchronously to wait for value to refresh)
+    setTimeout(() => this.allForms.controls.registerConfirmationPassword.updateValueAndValidity(), 0);
+  }
+
+  // validator for confirm password in sign up page
+  public confirmationValidator = (control: FormControl): { [s: string]: boolean } => {
+    if (this.allForms && control.value !== this.allForms.controls.registerPassword.value) {
+      return { confirm: true };
+    }
+    return {};
+  };
+
+  /**
+   * This method is respond for the sign in button in the pop up
+   * It will send data inside the text entry to the user service to login
```

Comment:
please fix the grammar here.  also, are we still using pop up? if not please update the comments.

Reply:
fixed.